src/iface/ethernet.rs

```diff
@@ -637,3 +637,168 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::Packet;
+    use phy::Loopback;
+    use wire::{ArpRepr, ArpOperation};
+    use wire::{EthernetAddress, EthernetFrame};
+    use wire::{IpAddress, IpCidr, Ipv4Address};
+    use iface::{ArpCache, SliceArpCache, EthernetInterface};
+    use socket::SocketSet;
+
+    #[test]
+    fn no_icmp_to_broadcast() {
+        // Unknown Ipv4 Protocol
+        //
+        // Because the destination is the broadcast address
+        // this should not trigger and Destination Unreachable
+        // response. See RFC 1122 § 3.2.2.
+        let bytes = vec![
+            // Ethernet Frame
+            0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
```

Comment:
Since we are not testing the packet parsing/construction routines here, do you think you could use `EthernetFrame`/`IpRepr`/`ArpRepr`/etc to construct the bytewise representations? That would make tests much easier to understand and write.

Reply:
This would definitely be feasible for the ARP tests. I'm not sure about how feasible it would be for this test since the protocol type is a protocol that isn't in `IpProtocol`.

Reply:
`IpProtocol::Unknown(0xwhatever)`